pkg/issuer/acme/http/http_test.go

```diff
@@ -107,12 +107,15 @@ func TestReachabilityCustomDnsServers(t *testing.T) {
 		t.Fatalf("Failed to resolve %s: %v", u.Host, err)
 	}
 
+	dnsServerStarted := make(chan struct{})
 	dnsServerCalled := int32(0)
 
-	server := &dns.Server{Addr: "127.0.0.1:15353", Net: "udp"}
+	server := &dns.Server{Addr: "127.0.0.1:15353", Net: "udp", NotifyStartedFunc: func() { close(dnsServerStarted) }}
 	defer server.Shutdown()
 
-	dns.HandleFunc(".", func(w dns.ResponseWriter, r *dns.Msg) {
+	mux := &dns.ServeMux{}
```

Comment:
We create a new mux here instead of using the default/ global mux, so that we cannot interfere with other mux registrations (in other tests or when running this test in parallel).